### PR TITLE
bt-214: add dropdown common component

### LINKED ISSUE #214: Add Dropdown common component
https://mui.com/base-ui/react-menu/

- [x] add common component
- [x] update header to add Sign out functional

Acceptance criteria for Dropdown common component

- Clicking on a closed field opens a menu of options
- Field containers should vertically align with other form components on a page
- A Dropdown open down 
- To select an option the user can click anywhere inside an option container
- Selecting an option from the menu closes it and the selected option text replaces the placeholder text in the field and also remains as an option in place if the menu is open
- Click outside of the option container closes it, and the field stays empty

Acceptance criteria for Sign out component

- after clicking on the registered user's avatar, a dropdown list appears
- dropdown list contains an option to Sign out
- criteria of the dropdown list “Sign out” is the same as the common component criteria
- after choosing “Sign out” the user is redirected to the page “Sign In”, the user is not logged in

### PR DESCRIPTION
**Pull request by JemiJe**
## notes
- changed styles will affect on [sidebar](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/pull/114) and pagelayout component
- as header `position: fixed` all elements need to know their margin-top, so I added --header-height to variables
- menu items can set in header-user-menu.tsx
- log out handles via redux action in auth

Screenshot (642)
Screenshot (648)

### imports
```jsx
import { Dropdown, Menu, MenuButton, MenuItem } from '../components.js';
```
### header
```jsx
            <Header
                avatarUrl="https://cdn.pixabay.com/photo/2023/07/27/13/37/cottage-8153413_1280.jpg"
                isOnline={true}
            />
```
### dropdown
```jsx
            <Dropdown {/*  open */}>
                <MenuButton>
                    <HeaderAvatar avatarUrl={avatarUrl} isOnline={isOnline} />
                </MenuButton>
                <Menu>
                    <MenuItem>
                        <Contrast fontSize="small" />
                        <Typography variant="h6">Another item</Typography>
                    </MenuItem>
                    <MenuItem>
                        <Settings fontSize="small" />
                        <Typography variant="h6">Settings</Typography>
                    </MenuItem>
                    <Divider />
                    <MenuItem>
                        <Logout fontSize="small" className={styles.signOutIcon} />
                        <Typography variant="h6" className={styles.signOut}>
                            Sign Out
                        </Typography>
                    </MenuItem>
                </Menu>
            </Dropdown>
```

### CODE REVIEW

**Comment by JosuerBague** [comment]
I merged development into this branch.

@JemiJe  There is a duplicate declaration for variable --header-height. 

One in line 66 and one line 107.

File: frontend/src/assets/css/variables.scss [hunk 1]
```diff
@@ -63,7 +63,7 @@
     --error-primary: var(--torch-red);
 
     // Header
-    --header-height: 60px;
+    --header-height: 72px;
```
Comment (nikita-remeslov) [inline frontend/src/assets/css/variables.scss:66]: we don't broke something with this change?

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/assets/css/variables.scss (hunk 1 above)
Comment (JemiJe) [reply frontend/src/assets/css/variables.scss:66]: definitely not, I contacted with @Simpsonkk and he tested new value as me too. and according page layout padding styles also was changed by this new varible frontend/src/bundles/common/components/page-layout/styles.module.scss
that's height we have in initial mockup
all main content of each route page is under mainContent, which will have padding `padding: var(--header-height) 0 0;` and not a magic number like previous `padding: 61px 0 0;`. I have already tested `/talent` route for example.

File: frontend/src/bundles/common/components/dropdown/components/menu/styles.module.scss [hunk 2]
```diff
@@ -0,0 +1,13 @@
+.menu {
+    z-index: 1;
+}
+
+.menu ul {
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/dropdown/components/menu/styles.module.scss]: https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/src/javascript.md - B10

File: frontend/src/bundles/common/components/header/components/header-user-menu/header-user-menu.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                message: 'You are logged out',
+            }),
+        );
+        navigate(AppRoute.SIGN_IN);
+    }, [dispatch, navigate]);
+
+    return (
+        <Menu>
+            {/* another items */}
+            {/* <Divider /> */}
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/header/components/header-user-menu/header-user-menu.tsx]: ?

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Review by MChornomaz — APPROVED** [review]

**Comment by nikita-remeslov** [comment]
Conflicts

**Review by ptarho — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**